tasks/devkit.rake

```diff
@@ -15,17 +16,35 @@ ENV['PATH'] = TDK::OS.join_env_path(
 
 PLAN_FILE = 'plan.tfplan'.freeze
 
-def destroy_if_fails(env)
+def destroy_if_fails(env, task)
   yield
-rescue StandardError => e
+rescue Exception => e
   puts "ERROR: #{e.message}"
   puts e.backtrace.join("\n")
-  task('destroy').invoke(env.name) if env.local_backend?
+  Rake::Task[task_in_current_namespace('destroy', task)].invoke(env.name) if env.local_backend?
   raise
 end
 
+def invoke(task_name, task_context, env, safe_invoke = false)
```

Comment:
What about doing

```
should_invoke = !safe_invoke || Rake::Task.task_defined?(task_name)
Rake::Task[task_name].invoke(env) if should_invoke
```

In this way, we can just ditch `invoke_if_defined` completely.